File: design-patterns/java/facade/TestingFramework.java
public class TestingFramework {
    public boolean runUnitTests() {
        System.out.println("Testing: Running unit tests...");
        simulateDelay(1500);
        System.out.println("Testing: Unit tests passed.");
        return true;
    }

    public boolean runIntegrationTests() {
        System.out.println("Testing: Running integration tests...");
        simulateDelay(3000);
        System.out.println("Testing: Integration tests passed.");
        return true;
    }

    private void simulateDelay(int ms) {
        try {
            Thread.sleep(ms);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}

File: design-patterns/csharp/facade/TestingFramework.cs
using System;
using System.Threading;

namespace Facade
{
    public class TestingFramework
    {
        public bool RunUnitTests()
        {
            Console.WriteLine("Testing: Running unit tests...");
            SimulateDelay(1500);
            Console.WriteLine("Testing: Unit tests passed.");
            return true;
        }

        public bool RunIntegrationTests()
        {
            Console.WriteLine("Testing: Running integration tests...");
            SimulateDelay(3000);
            Console.WriteLine("Testing: Integration tests passed.");
            return true;
        }

        private void SimulateDelay(int ms)
        {
            Thread.Sleep(ms);
        }
    }
} 